### DISCUSSION TITLE
Powershell version of SWI related embedding script

### DISCUSSION THREAD

**Discussion by hnbeck** [LogtalkDotOrg/logtalk3#142] [Announcements]
Hi all, 

I'm using Logtalk a lot with Win11 and Win10. I've started to port some SWI and embedding related  scripts to PowerShell. They can be found  [here](https://github.com/hnbeck/ECLogtalkTools/tree/main/Scripts).

These are EXPERIMENTAL at the moment, but the swipl_logtalk_qlf.ps1 should do all (in the Windows way) as the original bash script. I've tested this for *my* setup, the tests of every execution branch in the script is limited :)

My setup is Win10, Win11, Logtalk 3.54 and 3.55 beta.

The setup script for LOGTALKUSER  doesn't work because I can not run WSH for security reasons.  So this one and some scripts related to Logtalk developer tools may be good next candidates for porting. 

Cheers

Hans

**Comment by pmoura** [comment]
It should be quite doable to write a PowerShell version of the `logtalk_user_setup.js` script as its mainly creating directories, copying files, and creating symbolic links/shortcuts.

**Reply by pmoura** [reply]
Writing and testing `logtalk_user_setup.ps1`. Hope to complete it today.

**Reply by hnbeck** [reply]
Hui, great :)

**Reply by pmoura** [reply]
I have a first version of the script working. Two pain points are (1) the script execution policy must be changed (I used `powershell.exe -ExecutionPolicy Unrestricted`) and (2) the script requires admin privileges (due to the creation of symbolic links).

**Comment by pmoura** [comment]
Pushed experimental versions of some of the integration and setup scripts. Feedback most welcome.

**Reply by pmoura** [reply]
Update the integration scripts to call the `logtalk_user_setup` when required.

**Comment by pmoura** [comment]
Pushed a working SWI-Prolog `swipl_logtalk_qlf.ps1` embedding script. Be sure to read the `scripts/embedding/NOTES.md` file for details.

**Comment by pmoura** [comment]
> The setup script for LOGTALKUSER doesn't work because I can not run WSH for security reasons. So this one and some scripts related to Logtalk developer tools may be good next candidates for porting.

Execution of PowerShell scripts is also disabled by default for security reasons. So, what's different in this regard from WSH scripts?

**Comment by pmoura** [comment]
The Windows installer for the latest git version ([download](https://ci.appveyor.com/project/pmoura/logtalk3/build/artifacts)) makes all `*.ps1` scripts available from the system path. Be sure to read the embedding [notes](https://github.com/LogtalkDotOrg/logtalk3/blob/master/scripts/embedding/NOTES.md) on how to use the scripts on Windows.

**Comment by pmoura** [comment]
Committed `lgt2svg.ps1` script. It converts the `*.dot` files generated by the `diagrams` tool to `*.svg` files.

**Comment by pmoura** [comment]
Committed PowerShell versions of all the `lgtdoc` tool shell scripts.

**Comment by pmoura** [comment]
Committed PowerShell version of the `logtalk_allure_report` script.

**Comment by pmoura** [comment]
Committed PowerShell version of the `logtalk_tester` script working (minus the timeout option). It requires PowerShell 7.2.x and manually enabling an experimental feature using the command:

```text
Enable-ExperimentalFeature PSNativeCommandArgumentPassing
```

**Reply by pmoura** [reply]
Timeout option implemented. It requires Git for Windows to be installed in its default location.
